torch/_prims/nvfuser_executor.py

```diff
@@ -268,11 +268,39 @@ def __call__(self, *args):
         )
 
 
+def _remove_empty_like_fill(gm: GraphModule):
+    # Remove empty_like + fill nodes that prevent lowering to nvprims
+    # This is a workaround for nonoptimal traces of C++ code `(1 - tensor)`
+    # https://github.com/pytorch/pytorch/issues/86612
+
+    # Here when we see a `sub` node, we check if the first input is a result of
+    # filling a tensor with a scalar
+    # If so, we replace the first argument of the `sub` node with a scalar
+    gm = deepcopy(gm)
+    for node in gm.graph.nodes:
+        if node.op == "call_function":
+            if node.target == torch.ops.aten.sub.Tensor:
+                # check if the first argument is a fill
+                if (
+                    node.args[0].op == "call_function"
+                    and node.args[0].target == torch.ops.aten.fill.Scalar
+                ):
+                    # Replace the first argument with the second argument of fill
+                    # aten.fill.Scalar(tensor, scalar)
+                    fill_node = node.args[0]
+                    scalar = fill_node.args[1]
+                    node.args = (scalar, *node.args[1:])
```

Comment:
do you rely on implicit scalar wrapping here? `scalar` argument won't match  `torch.ops.aten.sub.Tensor` schema

Reply:
Yes, I'm relying on this line:
https://github.com/pytorch/pytorch/blob/6ee94b572ac248b11a90558b7358c242ad9b56fa/torch/csrc/jit/python/pybind_utils.cpp#L93
```py
In [1]: import torch

In [2]: a = torch.randn(3, 3, device='cuda')

In [3]: torch.ops.aten.sub.Tensor(1, a)
Out[3]:
tensor([[ 1.2669,  1.2413,  1.5390],
        [ 1.5367,  1.2781,  0.6397],
        [ 0.5167,  2.3106, -0.4991]], device='cuda:0')
```